.circleci/config.yml

```diff
@@ -49,8 +54,30 @@ jobs:
           name: Verify App Store Target Builds
           command: bundle exec fastlane test_app_store_build
 
+  App Store Upload:
+    executor: *xcode_image
+    steps:
+      - setup_environment
+      - run:
+          name: Install Sentry CLI
+          command: brew install getsentry/tools/sentry-cli
+      - run:
+          name: Build and Upload to App Store and GitHub
+          command: |
+            bundle exec fastlane build_and_upload_app_store create_github_release:true
+
 workflows:
   simplenote_macos:
+    when:
+      not: << pipeline.parameters.app_store_build >>
     jobs:
       - Test
       - Verify App Store Target Builds
+
+  app_store:
+    when: << pipeline.parameters.app_store_build >>
+    jobs:
+      - Test
```

Comment:
Running the tests at this step may be redundant in most cases. But, because we now trigger this via Fastlane locally, there is no hard guarantee that no new change made it into the codebase.

I think the extra peace of mind is a price worth paying the time delay.